e2e/cypress/integration/signin_authentication/authentication_spec.js

```diff
@@ -0,0 +1,283 @@
+        cy.get('#accountSettingsModal').should('exist').within(() => {
+            // Click "Notifications"
+            cy.findByText('Notifications').should('be.visible').click();
+
+            // Click "Desktop"
+            cy.findByText('Desktop Notifications').should('be.visible').click();
+
+            // # Set your desktop notifications to Never
+            cy.get('#desktopNotificationNever').check();
+
+            // Click "Save"
+            cy.findByText('Save').scrollIntoView().should('be.visible').click();
+
+            // Close the modal.
+            cy.get('#accountSettingsHeader').find('button').should('be.visible').click();
+        });
+
+        cy.visit(`/${testTeam.name}/channels/town-square`).wait(timeouts.ONE_SEC);
+
+        // Visit the town-square with the notification API stubbed.
+        spyNotificationAs('withNotification', 'granted');
+
+        // # From a separate browser session, login to the same server as the same user
+        // # Click the hamburger menu and select Account Settings ➜ Security
+        // # Click "View and Logout of Active Sessions", then find and close the session created in step 1
+        // Since we are testing this on browser, we can revoke sessions with admin user.
+        const sysadmin = getAdminAccount();
+        cy.externalRequest({user: sysadmin, method: 'post', path: `users/${testUser.id}/sessions/revoke/all`}).wait(timeouts.HALF_MIN);
+
+        // # Go back and view the original session app/browser, and wait until you see a desktop notification (may take up to a minute)
+        // * Desktop notification is sent (may take up to 1 min)
+        //cy.get('@withNotification').should('have.been.calledOnce');
+
+        // * Login page shows a message above the login box that the session has expired.
+        cy.get('#login_section .alert-warning').should('contain.text', 'Your session has expired. Please log in again.');
+    });
+});
+
+function getUserSecret(user) {
+    return cy.url().then((url) => {
+        if (url.includes('mfa/setup')) {
+            return cy.get('#mfa').wait(timeouts.HALF_SEC).find('.col-sm-12').then((p) => {
+                const secretp = p.text();
+                const secret = secretp.split(' ')[1];
+
+                const token = authenticator.generateToken(secret);
+                cy.get('#mfa').find('.form-control').type(token);
+                cy.get('#mfa').find('.btn.btn-primary').click();
+
+                cy.wait(timeouts.HALF_SEC);
+                cy.get('#mfa').find('.btn.btn-primary').click();
+                return cy.wrap({secret});
+            });
+        }
+
+        // # If the user already has MFA enabled, reset the secret.
+        return cy.apiGenerateMfaSecret(user.id).then((res) => {
+            const secret = res.code.secret;
+            return cy.wrap({secret});
+        });
+    });
+}
+
+function fillCredentialsForUser(user) {
+    cy.findByPlaceholderText('Email or Username').clear().type(user.username);
```

Comment:
Failed on me when running against server license, due to LDAP enabled. Either disable LDAP or use `cy.get('#loginId')` which I'm fine in this case.